### PR TITLE
refactor: [M3-7720] - Refactor AccessKeyTable - Eliminate react anti patterns

### PR DESCRIPTION
**Pull request by cpathipa**
## Description 📝

- Refactors Object-Storage - AccessKeyTable component.
- Eliminates react anti patterns like calling functions that render JSX.
- No Visual changes.

## How to test 🧪
(How to verify changes)
- Validate Object storage Access key flow - create/update/revoke.
- Validate the Access Key landing page and ensure there are no regressions in the Access Key flow.

## As an Author I have considered 🤔

*Check all that apply*

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [ ] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

**Review by abailly-akamai — COMMENTED** [review]
can we fix this markup error as well?

Screenshot 2024-01-30 at 15 29 29

**Comment by abailly-akamai** [comment]
@cpathipa `/object-storage/access-keys` is crashing with the `OBJ Multi-Cluster` flag `on`

File: packages/manager/src/features/ObjectStorage/AccessKeyLanding/AccessKeyTable/AccessKeyActionMenu.tsx [hunk 1]
```diff
@@ -18,7 +18,7 @@ interface Props {
   openRevokeDialog: (key: ObjectStorageKey) => void;
 }
 
-export const AccessKeyMenu = (props: Props) => {
+export const AccessKeyActionMenu = (props: Props) => {
```
Comment (carrillo-erik) [inline packages/manager/src/features/ObjectStorage/AccessKeyLanding/AccessKeyTable/AccessKeyActionMenu.tsx:21]: Just a small nitpick, the props could have been destructured in the argument to be consistent with the other files.

**Comment by carrillo-erik** [comment]
I was able to verify that I can create/update/revoke access keys after the changes. One thing I also noticed was that there are associated events with the actions I've mentioned, however, we do not show any toast notifications.

**Review by carrillo-erik — APPROVED** [review]

**Review by abailly-akamai — APPROVED** [review]

File: packages/manager/src/features/ObjectStorage/AccessKeyLanding/AccessKeyTable/AccessKeyActionMenu.tsx (hunk 1 above)
Comment (cpathipa) [reply packages/manager/src/features/ObjectStorage/AccessKeyLanding/AccessKeyTable/AccessKeyActionMenu.tsx]: Done - [3a10ba7](https://github.com/linode/manager/pull/10124/commits/3a10ba71d7c31a25c906b10ce598eeec798880b2)

**Pull request merged**